include/astarte_device_sdk/mqtt/exceptions.hpp

```diff
@@ -0,0 +1,94 @@
+// (C) Copyright 2025, SECO Mind Srl
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef ASTARTE_MQTT_EXCEPTIONS_HPP
+#define ASTARTE_MQTT_EXCEPTIONS_HPP
+
+/**
+ * @file exceptions.hpp
+ * @brief Hierarchy of custom exceptions used throughout the Astarte device SDK.
+ */
+
+#pragma once
```

Comment:
It makes little sense to use both #pragma once and the #ifndef as they serve the same purpose.
Keep the ifdef to be coherent with the other headers.